src/lib.rs

```diff
@@ -42,59 +48,182 @@ fn parse_line(line: &str, skip: usize) -> &str {
+    /// This should be a string consisting only of 0-9, a-z, ".", "_", "-" (no upper case letters).
     ///
-    /// **IE:** `18.04 LTS (Bionic Beaver)`
-    pub version_id:         String,
-    /// The version of this OS release.
+    /// This field is optional.
     ///
     /// **IE:** `18.04`
-    pub version:            String,
+    pub version_id: String,
+
+    /// A pretty name for this OS which can be presented to the user.
+    ///
+    /// Defaults to `Linux`.
+    ///
+    /// **IE:** `Pop!_OS 18.04 LTS`
+    pub pretty_name: String,
+
+    /// The suggested presentation color when showing the OS name in the console.
+    /// Given in a format suitable for inclusion in the ESC [ m ANSI/ECMA-48 escape code.
+    ///
+    /// This field is optional.
+    /// Since this field is optional, it may be "" (don't forget to check before printing).
+    ///
+    /// **IE:** `1;31` (red) or `38;2;23;147;209` (RGB 23, 147, 209)
+    pub ansi_color: String,
+
+    /// A CPE name for the operating system, in URI binding syntax.
+    /// See the Common Platform Enumeration Specification.
+    ///
+    /// This field is optional.
+    ///
+    /// **EG:** `cpe:/o:fedoraproject:fedora:17`
+    pub cpe_name: String,
+
+    /// The homepage of this OS.
+    ///
+    /// This field is optional.
+    pub home_url: String,
+    /// The documentation page of this OS.
+    ///
+    /// This field is optional.
+    pub documentation_url: String,
+    /// The URL for seeking support with this OS release.
+    ///
+    /// This field is optional.
+    pub support_url: String,
+    /// The URL where bugs should be reported for this OS.
+    ///
+    /// This field is optional.
+    pub bug_report_url: String,
+    /// The URL describing this OS's privacy policy.
+    ///
+    /// This field is optional.
+    pub privacy_policy_url: String,
+
+    /// A unique ID for the image used as the origin for the OS (it is not updated).
+    ///
+    /// This field is optional.
+    pub build_id: String,
+
+    /// The variant or edition of the OS.
+    ///
+    /// This field is optional.
+    ///
+    /// **EG:** `Server`
+    pub variant: String,
+
+    /// An ID for the variant or edition of the OS.
+    /// This should be a string consisting only of 0-9, a-z, ".", "_", "-" (no upper case letters).
+    ///
+    /// This field is optional.
+    pub variant_id: String,
+
+    /// The name of the logo for the operating system, as defined by freedesktop.org Icon Theme
+    /// Specification.
+    ///
+    /// This field is optional.
+    pub logo: String,
+
     /// Additional keys not covered by the API.
-    pub extra:              BTreeMap<String, String>
+    pub extra: BTreeMap<String, String>,
+}
+
+impl Default for OsRelease {
+    fn default() -> OsRelease {
+        OsRelease {
+            name: "Linux".into(),
+            id: "linux".into(),
+            pretty_name: "Linux".into(),
+
+            version: String::new(),
```

Comment:
(I know this is an ancient PR, but we were considering adding a new use of this crate in a project and I went to look at it and noticed the last commit was 6 years ago, which...isn't wrong since obviously things are stable, but on the other hand I'd like to just know things I depend on are maintained)

Anyways: I think this can just be `..Default::default()` right?

Reply:
Haven't seen any use case that required updating it. There is already a hash map storing extra fields.